### PR TITLE
Patch for MbedTLS PR#4978 - fixes bignum multiplication for newer versions of clang (Xcode 13/14)

### PR DESCRIPTION
**Pull request by the-real-adammork**
### Description

Newer versions of `clang` that ships with Xcode 13 (and up) cause problems with inline-assembly code in MbedTLS.  The [original issue](https://github.com/Mbed-TLS/mbedtls/issues/4943) says this:

> Some of our inline assembly code is missing annotations telling the compiler what the assembly reads and writes (register, memory). On the platforms we test with, there are generally enough annotations to get correct code, but we're at the mercy of a new compiler version that optimizes better. This happened on x86_64 with Clang 12 and the bignum multiplication code (https://github.com/Mbed-TLS/mbedtls/issues/4116, https://github.com/Mbed-TLS/mbedtls/issues/4786, https://github.com/Mbed-TLS/mbedtls/issues/4917) ...

My PR adds the code fix for 2.x that came in this [MbedTLS PR 4978](https://github.com/Mbed-TLS/mbedtls/pull/4978/files). 

### Additional Notes

MobileCoin's [IAS Report Verifier](https://github.com/mobilecoinfoundation/mobilecoin/blob/master/attest/verifier/src/ias.rs#L140) uses this code to verify the certificate chain. The bug causes this [line](https://github.com/mobilecoinfoundation/mobilecoin/blob/master/attest/verifier/src/ias.rs#L140) to fail with an `  
RsaVerifyFailed = ERR_RSA_VERIFY_FAILED` MbedTLS error which stops the Attestation process. On Swift this shows up as a `ReportVerification(BadSignature)` error.

### CODE REVIEW

File: mbedtls-sys/vendor/ChangeLog.d/muladdc-amd64-memory.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Bugfix
```
Comment (nick-mobilecoin) [inline mbedtls-sys/vendor/ChangeLog.d/muladdc-amd64-memory.txt]: :wrench: It looks like this got removed in https://github.com/Mbed-TLS/mbedtls/pull/4978 and was consolidated to `muladdc-memory.txt`

**Review by nick-mobilecoin — APPROVED** [review]
LGTM

**Review by eranrund — APPROVED** [review]

File: mbedtls-sys/vendor/ChangeLog.d/muladdc-amd64-memory.txt (hunk 1 above)
Comment (the-real-adammork) [reply mbedtls-sys/vendor/ChangeLog.d/muladdc-amd64-memory.txt]: good catch

**Comment by the-real-adammork** [comment]
@eranrund @nick-mobilecoin I fixed my commits to be signed/verified, can one of you do the merge for me ?

**Pull request merged**